### PR TITLE
fix: skip rebuild if using the tools command

### LINKED ISSUE #24551: the `./kc.sh tools completion` command cannot be recognized correctly
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/cli

### Describe the bug

When I want to generate bash/zsh completion script by `./kc.sh tools completion`, an error occurred with the following log:

```
Changes detected in configuration. Updating the server image.
Unknown option: 'completion'
Try 'kc.sh build --help' for more information on the available options.
```

However I can see the help msg of this completion command with `./kc.sh tools --help`, so I think the command I typed is correct.

I tried to use historical versions of binaries and found：

* 21.1.2, this completion command works fine
* 22.0.0, this completion command doesn't work, the same error as above

I tried to find the cause of the problem but couldn't find it. I hope I can get help here.

### Version

22.0.5

### Expected behavior

command `./kc.sh tools completion` works fine

### Actual behavior

command `./kc.sh tools completion` say `Unknown option: 'completion'`

### How to Reproduce?

unpack the binary tar.gz package, run the above command.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by shawkins**
closes #24551

### CODE REVIEW

File: quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ToolsCommandDistTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import io.quarkus.test.junit.main.Launch;
+import io.quarkus.test.junit.main.LaunchResult;
+
+import org.junit.jupiter.api.Test;
+import org.keycloak.it.junit5.extension.DistributionTest;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@DistributionTest
+public class ToolsCommandDistTest {
```
Comment (vmuzikar) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ToolsCommandDistTest.java:29]: Should we perhaps also add a test similar to this?
https://github.com/keycloak/keycloak/blob/71b6757c2f7a3e08b857944bf9416f93602318a1/quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/HelpCommandDistTest.java#L153-L167

Comment (shawkins) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ToolsCommandDistTest.java:29]: @vmuzikar isn't that dependent upon changes to the shell scripts?  I thought from our other discussion we should move away from putting additional logic in the scripts.

Comment (mabartos) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ToolsCommandDistTest.java:29]: @vmuzikar Could you please describe what you mean, as the `tools` is a separated command with one single sub-command?

Comment (vmuzikar) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/ToolsCommandDistTest.java:29]: You are right, that's additional enhancement with added complexity. Looks good as is then. Thanks.

**Review by vmuzikar — APPROVED** [review]

**Comment by vmuzikar** [comment]
@shawkins This is probably a good candidate for backport.

**Pull request merged**